client/session_test.go

```
@@ -0,0 +1,79 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	v2netmap "github.com/nspcc-dev/neofs-api-go/v2/netmap"
+	"github.com/nspcc-dev/neofs-api-go/v2/rpc/client"
+	"github.com/nspcc-dev/neofs-api-go/v2/session"
+	neofscrypto "github.com/nspcc-dev/neofs-sdk-go/crypto"
+	"github.com/nspcc-dev/neofs-sdk-go/crypto/test"
+	"github.com/stretchr/testify/require"
+)
+
+type emptySessionParameter struct {
+	signer  neofscrypto.Signer
+	setBody func(body *session.CreateResponseBody)
+}
+
+func (m emptySessionParameter) netMapSnapshot(context.Context, v2netmap.SnapshotRequest) (*v2netmap.SnapshotResponse, error) {
+	return nil, nil
+}
+
+func (m emptySessionParameter) createSession(*client.Client, *session.CreateRequest, ...client.CallOption) (*session.CreateResponse, error) {
+	var body session.CreateResponseBody
+	m.setBody(&body)
+
+	var resp session.CreateResponse
+	resp.SetBody(&body)
+
+	if err := signServiceMessage(m.signer, &resp); err != nil {
+		panic(fmt.Sprintf("signServiceMessage: %v", err))
```

why not return the err?

---

good point, replaced with error